File: 70-climbing-stairs/70-climbing-stairs.cpp
class Solution {
public:
    
    vector<int> dp = vector<int>(46,-1);
    
    int climbStairs(int n) {
        if(n <= 1) return 1;
        if(dp[n] != -1) return dp[n];
        
        //left and right here mean the RECURSIVE TREE'S left and right
        
        int left = climbStairs(n-1);
        int right = climbStairs(n-2);
        
        dp[n] = left+right;
        return left+right;
    }
};

File: 70-climbing-stairs/70-climbing-stairs.py
class Solution:
    dp = [-1]*46
    
    def climbStairs(self, n: int) -> int:
        if n<=1: return 1
        if self.dp[n] != -1: return self.dp[n]
        
        left = self.climbStairs(n-1)
        right = self.climbStairs(n-2)
        
        self.dp[n] = left+right
        return self.dp[n]